magicctapipe/scripts/lst1_magic/semi_automatic_scripts/new_nsb_DL1_to_DL2.py

```diff
@@ -0,0 +1,209 @@
+"""
+This script creates the bashscripts necessary to apply "lst1_magic_dl1_stereo_to_dl2.py"
+to the DL1 stereo data (real and MC). It also creates new subdirectories associated with
+the data level 2. The DL2 files are saved at:
+WorkingDirectory/DL2/
+and in the subdirectories therein.
+
+Usage:
+$ python new_DL1_to_DL2.py -c configuration_file.yaml
+
+"""
+import argparse
+import glob
+import logging
+import os
+import time
+from pathlib import Path
+import datetime 
+import joblib
+import numpy as np
+import pandas as pd
+import yaml
+from magicctapipe import __version__
+from magicctapipe.io import resource_file
+from magicctapipe.scripts.lst1_magic.semi_automatic_scripts.clusters import (
+    rc_lines,
+    slurm_lines,
+)
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+def ST_NSB_List(target_dir, nsb_list, nsb_limit, source):
+    """
+    This function creates the lists of runs separeted by run period and NSB level.
+
+    Parameters
+    ----------
+    target_dir: str
+        Path to the working directory
+    nsb_list:
+    nsb_limit:
+    	
+    source:
+    	source name
+    """ 
+
+    "NSB dataframe"
+    df_LST = pd.read_hdf(
+        "/fefs/aswg/workspace/elisa.visentin/auto_MCP_PR/observations_LST.h5",
+        key="joint_obs",
+    )
+
+    "Period Lists"
+    ST_list = ["ST0320A", "ST0319A", "ST0318A", "ST0317A", "ST0316A"]
+    ST_begin = ["2023_03_10", "2022_12_15", "2022_06_10", "2021_12_30", "2020_10_24"]
+    ST_end = ["2025_01_30", "2023_03_09", "2022_08_31", "2022_06_09", "2021_09_29"]
+    # ST0320 ongoing -> 'service' end date
+
+    "Loops over all runs of all nights"
+    Nights_list = np.sort(glob.glob(f"{target_dir}/v{__version__}/{source}/DL1Stereo/*"))
+    for night in Nights_list:
+        "Night period"
+        night_date=night.split('/')[-1]
+        print('night',night_date)
```

Comment:
Of course, when we manage to make the scripts work fine, we have to remove some of the 'useless' print